src/services/mongoose/schema.ts

```diff
@@ -348,7 +348,9 @@ const UserSchema = new Schema(
       unique: true,
       sparse: true,
     },
-    deletedPhone: { type: String },
+    deletedPhone: {
```

Comment:
I would prefer an immutable refactor rather than an update in place. Keep this field and add a new one `deletedPhones`.

Reply:
it's to simplify the state/code. more code/code path to maintain with your approach versus a "clean" upgrade. also simplify the data pipeline down the line.

Reply:
it also aims at making this PR more digestable: https://github.com/GaloyMoney/galoy/pull/2713 (ready to be review/merged btw). if we need to think about immutable refactor on top of it, then it's even more context to keep in your head